GPy/kern/src/stationary.py

```diff
@@ -307,6 +310,21 @@ def gradients_XX_diag(self, dL_dK_diag, X):
         l4 =  np.ones(X.shape[1])*self.lengthscale**2
         return dL_dK_diag * (np.eye(X.shape[1]) * -self.dK2_drdr_diag()/(l4))[None, :,:]# np.zeros(X.shape+(X.shape[1],))
         #return np.ones(X.shape) * d2L_dK * self.variance/self.lengthscale**2 # np.zeros(X.shape)
+    
+    def dgradients_dX(self, X, X2, dimX):
```

Comment:
Is it possible to chain these gradients through gradients_X?

Reply:
This would mean it would be pushed through to all stationary kernels, and not 'just' the rbf kernel? 
I have not looked at the maths exactly, but it should be possible to do that with less effort on the actualy kernel side, when first transforming the gradients in here to go through `r` and then applying the inner gradient in the respective kernel? As has been done in `gradients_X` with `self.dK_dr_via_X(X, X2)`?

Reply:
I did the maths before. The problem here is that the computations get slower + actually implementing the chain rule (especially for triple gradients dgradients2_dXdX2) is pain in the ass in stationary kernel. I did the maths and also have the code somewhere on my private repository but for my own applications it work better without chaining the gradients.

Reply:
So what do you think? I have no need for derivative observations for other stationary kernels at this point and thus no plans for computing and implementing them.

Reply:
would it be possible to just make `gradient` point to the `base_kern.gradient` in where necessary?

e.g.:

```python
@property
def gradient(self):
    return self.base_kern.gradient

@gradient.setter
def gradient(self, gradient):
   self.base_kern.gradient = gradient
```

this would mean, there are no gradients on the multioutput_derivative_kern itself, though, so maybe this needs further discussion?

Reply:
This would go in the `KernWrapper` and `MultioutputKernel`, so that you can just use the gradient itself? That should do the trick, I believe?

Reply:
Aah! Now I see what you mean! I'll try this out!